pandas/core/base.py

```diff
@@ -1032,11 +1032,8 @@ def nunique(self, dropna: bool = True) -> int:
         >>> s.nunique()
         4
         """
-        uniqs = self.unique()
-        n = len(uniqs)
-        if dropna and isna(uniqs).any():
-            n -= 1
-        return n
+        obj = remove_na_arraylike(self) if dropna else self
```

Comment:
Late comment, but would it be more performant if we removed the NAs *after* calculating the uniques? (assume the uniques is typically a much smaller array, and not this does an extra scan and creates a copy of the full array)